packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/PropertyProvider.cs

```diff
@@ -60,6 +77,58 @@ public PropertyProvider(InputModelProperty inputProperty)
             ExplicitInterface = explicitInterface;
             XmlDocs = GetXmlDocs();
             WireInfo = wireInfo;
+
+            InitializeParameter(Name, description ?? FormattableStringHelpers.Empty, Type, ParameterValidationType.None);
+            InitializeField(backingField);
+        }
+
+        [MemberNotNull(nameof(_parameter))]
+        private void InitializeParameter(string propertyName, FormattableString description, CSharpType propertyType, ParameterValidationType validation)
+        {
+            var parameterName = propertyName.ToVariableName();
+            _parameter = new(() => new ParameterProvider(parameterName, description, propertyType)
+            {
+                Validation = validation,
+                Property = this
+            });
+        }
+
+        private void InitializeField(FieldProvider? backingField)
+        {
+            if (backingField == null)
+                return;
+
+            _backingField = backingField;
+            backingField.Property = this;
+        }
+
+        private static ParameterValidationType GetParameterValidation(InputModelProperty property, CSharpType propertyType)
```

Comment:
we should be able to remove this since you can get the validation from the AsParameter